src/main/java/org/lscode/DupLocator/DupLocator.java

```diff
@@ -0,0 +1,246 @@
+/**
+ *  file:    DupLocator.java
+ *  desc:    the main class of the DupLocator package
+ *  author:  ls-code
+ *  license: GNU General Public License v3.0
+ */
+
+package org.lscode.DupLocator;
+
+import java.lang.reflect.Array;
+import java.util.*;
+
+public class DupLocator extends Observable implements Observer{
+
+    private MultiDigest digestGenerator;
+    private String[] paths;
+    private FileArray allFiles = new FileArray();
+    private FileStorage<String> dups;
+    private FileStorageMap<String, String> namesakes;
+    private List<String> dirs;
+    private List<String> failedDirs = new ArrayList<>();
+    private long filesProcessedTotal = 0;
+    private long filesProcessedByCurrentFinder = 0;
+    private Stage stage = Stage.NONE;
+    private Phase phase;
+    private Boolean processing = false;
+    private int dirsTotal = 0;
+    private int dirNo = 0;
+
+
+    public enum Stage {
+        NONE, FILES, DUPLICATES, NAMESAKES
+    }
+
+    public enum Phase {
+        START, INPROGRESS, RESULT, END
+    }
+
+    public DupLocator(MultiDigest digestGenerator, String[] paths){
+        this.paths = paths;
+        this.digestGenerator = digestGenerator;
+        dirsTotal = Array.getLength(paths);
+    }
+
+    public FileStorage<String> getDups(){
+        if (dups == null){
+            if (allFiles.isEmpty() ){
+                findFilesInAllDirs();
+            }
+            findDups();
+        }
+        return dups;
+    }
+
+    public FileStorageMap<String, String> getNamesakes(){
+        if (namesakes == null){
```

Comment:
to samo co wyzej

Do tego mysle ze ten if jest za bardzo nagniezdzony, pomysl o rozbiciu tego na inne metody